### PR TITLE
feat: add authentication

### PR DESCRIPTION
**Pull request by nerdyspook**
**Added Login & Signup Authentication**
> save user details locally

### CODE REVIEW

**Review by Prabhav-Sharma — APPROVED** [review]
Clean and easy to navigate code

File: .env [hunk 1]
```diff
@@ -0,0 +1 @@
+REACT_APP_JWT_SECRET=78tsgryb4i7ythxb4i747iuer
```
Comment (prathmeshjagtap) [inline .env:1]: don't push env file on github

**Review by prathmeshjagtap — APPROVED** [review]
The code is well structured, I liked that you are using regex for email checking also you have made a separate function handle change that's good. 

Just don't push env file on GitHub you can add it in git ignore

File: .env (hunk 1 above)
Comment (nerdyspook) [reply .env:1]: Thanks, I will do it now.

**Pull request merged**